certchain/certmanager/manager_test.go

```diff
@@ -45,16 +44,12 @@ func TestManager(t *testing.T) {
 	{
 		t.Log("Started with augm0409")
 
-		select {
-		case got := <-cache.OnWrite:
-			if diff := cmp.Diff(augm0409, got, certComparer); diff != "" {
-				t.Errorf("cache mismatch (-want +got):\n%s", diff)
-			}
-		case <-time.After(defaultTimeout):
-			t.Error("cache timeout")
+		got, _ := m.Cache.Read(augm0409.Digest)
```

Comment:
I think you need to wait for `<- cache.avail`. Maybe try (if you haven't already):

```bash
$ go test -count=1 -race ./certchain/certmanager
```

Reply:
I wait for <- cache.avail in Read().  I ran the test you pasted and it works:

banaag@banaag:~/github/webpackager$ go test -count=1 -race ./certchain/certmanager
ok      github.com/google/webpackager/certchain/certmanager     0.542s

Reply:
Ack.